### PR TITLE
Inline most `{BugCheckerRefactoring,Compilation}TestHelper` fields

### PR DESCRIPTION
**Pull request by rickie**
As discussed in https://github.com/PicnicSupermarket/error-prone-support/pull/426#discussion_r1060477059.

### CODE REVIEW

**Comment by rickie** [comment]
Suggested commit message:
```
Inline `CompilationTestHelper`s that specify command-line arguments (#442)
```

**Review by Stephan202 — COMMENTED** [review]
Ah, I think we had a misunderstanding. We can simply inline all helpers that are used in only a single test, right? (And perhaps even if they are reused?)

**Review by chamil-prabodha — APPROVED** [review]

File: refaster-runner/src/test/java/tech/picnic/errorprone/refaster/runner/RefasterTest.java [hunk 1]
```diff
@@ -25,41 +25,32 @@
...
-              containsPattern(
-                  "\\[Refaster Rule\\] FooRules\\.ExtraGrouping\\.StringOfSizeTwoRule: "
-                      + "A custom subgroup description\\s+.+\\s+"
-                      + "\\(see https://example.com/rule/FooRules#ExtraGrouping.StringOfSizeTwoRule\\)"))
-          .expectErrorMessage(
-              "StringOfSizeThreeRule",
-              containsPattern(
-                  "\\[Refaster Rule\\] FooRules\\.ExtraGrouping\\.StringOfSizeThreeRule: "
-                      + "A custom description about matching three-char strings\\s+.+\\s+"
-                      + "\\(see https://example.com/custom\\)"));
```
Comment (Stephan202) [inline refaster-runner/src/test/java/tech/picnic/errorprone/refaster/runner/RefasterTest.java:52]: This one is now duplicated :grimacing:

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/util/ThirdPartyLibraryTest.java [hunk 2]
```diff
@@ -18,12 +18,9 @@
 import reactor.core.publisher.Flux;
 
 final class ThirdPartyLibraryTest {
-  private final CompilationTestHelper compilationTestHelper =
-      CompilationTestHelper.newInstance(TestChecker.class, getClass());
```
Comment (Stephan202) [inline error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/util/ThirdPartyLibraryTest.java:22]: Here 100% of tests use this field; I'd keep this one as-is.

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/StringCaseLocaleUsageTest.java [hunk 3]
```diff
@@ -54,7 +48,7 @@ void identification() {
 
   @Test
   void replacementFirstSuggestedFix() {
-    refactoringTestHelper
+    newInstance(StringCaseLocaleUsage.class, getClass())
```
Comment (Stephan202) [inline error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/StringCaseLocaleUsageTest.java:51]:
```suggestion
    BugCheckerRefactoringTestHelper.newInstance(StringCaseLocaleUsage.class, getClass())
```
Likewise below.

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/PrimitiveComparisonTest.java [hunk 4]
```diff
@@ -5,22 +5,16 @@
...
-      BugCheckerRefactoringTestHelper.newInstance(PrimitiveComparison.class, getClass());
-
-  // XXX: There are no tests for multiple replacements within the same expression:
-  // - Error Prone doesn't currently support this, it seems.
-  // - The `BugCheckerRefactoringTestHelper` throws an exception in this case.
-  // - During actual compilation only the first replacement is applied.
-  // XXX: Can we perhaps work-around this by describing the fixes in reverse order?
-
-  // The logic for `char` and `short` is exactly analogous to the `byte` case.
+  /* The logic for `char` and `short` is exactly analogous to the `byte` case. */
```
Comment (Stephan202) [inline error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/PrimitiveComparisonTest.java:14]:
```suggestion
  /** @implNote The logic for `char` and `short` is exactly analogous to the `byte` case. */
```
Perhaps.

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/RedundantStringConversionTest.java [hunk 5]
```diff
@@ -10,19 +10,9 @@
 // `String.valueOf(null)` may not. That is because the latter matches `String#valueOf(char[])`. We
 // could special-case `null` arguments, but that doesn't seem worth the trouble.
 final class RedundantStringConversionTest {
```
Comment (Stephan202) [inline error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/RedundantStringConversionTest.java:12]: This one is an edge case, w.r.t. field reuse :thinking:

**Comment by rickie** [comment]
The only one where it would make the test clearer is the `RefasterTest`. Now that test is a bit too big if you ask me. WDYT?

**Review by Stephan202 — COMMENTED** [review]
Quick Github UI-based review :)

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/StringCaseLocaleUsageTest.java (hunk 3 above)
Comment (rickie) [reply error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/StringCaseLocaleUsageTest.java:51]: Started working on that upstream fix. Will do that after applying this feedback.

**Review by rickie — COMMENTED** [review]
Added a commit with all feedback fixed.

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/StringCaseLocaleUsageTest.java (hunk 3 above)
Comment (rickie) [reply error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/StringCaseLocaleUsageTest.java]: Opened: https://github.com/PicnicSupermarket/error-prone/pull/58.

File: error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/FluxFlatMapUsageTest.java [hunk 6]
```diff
@@ -73,7 +67,7 @@ void identification() {
 
   @Test
   void replacementFirstSuggestedFix() {
-    refactoringTestHelper
+    newInstance(FluxFlatMapUsage.class, getClass())
```
Comment (chamil-prabodha) [inline error-prone-contrib/src/test/java/tech/picnic/errorprone/bugpatterns/FluxFlatMapUsageTest.java:70]: We can apply the same for the method below as well.
```suggestion
    BugCheckerRefactoringTestHelper.newInstance(FluxFlatMapUsage.class, getClass())
```

**Review by Stephan202 — APPROVED** [review]
Added one more commit. Nice!

Suggested commit message:
```
Inline most `CompilationTestHelper` and `BugCheckerRefactoringTestHelper` fields (#442)
```

**Comment by rickie** [comment]
Made a tweak to the suggested commit message and sorted the two entries. This fits the number of characters, WDYT?

**Comment by Stephan202** [comment]
Works for me :+1:
